### ISSUE TITLE
Regression in fault tolerance for incorrect photometric interpretation of YBR_FULL_422 for pixel data encap as JPEG lossless RGB

### ISSUE THREAD

**Issue by justinfalk** [dcm4che/dcm4che#543]
Following up on Google Groups post:  https://groups.google.com/forum/#!topic/dcm4che/6AMSke72zsk

It seems that a JPEG lossless (1.2.840.10008.1.2.4.70) image with RGB encapsulated data with incorrect photometric interpretation of YBR_FULL_422 is no longer able to be decoded with 5.16.1 and later versions of dcm4che3.  This works in dcm4che2 and earlier versions of dcm4che3.  

There error received is:
`java.lang.IllegalArgumentException: Raster ByteInterleavedRaster: width = 1020 height = 1320 #numDataElements 3 dataOff[0] = 2 is incompatible with ColorModel ColorModel: #pixelBits = 24 numComponents = 3 color space = org.dcm4che3.image.YBRColorSpace@40ea8dc2 transparency = 1 has alpha = false isAlphaPre = false`

**Comment by nroduit** [comment]
The problem is that the syntax (1.2.840.10008.1.2.2.4.70) allows several color models and that in general the management of decompression, compression, and transcoding according to the color models, lossless and lossy syntax, and the different pixel structures is really complicated. 

As a result, adding the management of all errors becomes totally impossible. However, when it is possible I have already added exceptions due to recurring manufacturer errors (see this [code part](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-imageio-opencv/src/main/java/org/dcm4che3/opencv/NativeImageReader.java#L306)). FYI, the current version fixes a long list of problems related to the decoder in relation to dcm4che2.

I'll try to look at the next time I update the library.

**Comment by nroduit** [comment]
The problem is that RGB, YBR_FULL and PALETTE COLOR are possible values in a color JPEG-LOSSLESS image, see http://dicom.nema.org/medical/dicom/current/output/chtml/part05/sect_8.2.html#table_8.2.1-2

The color space transformation is not a part of JPEG. So it is crucial to set the right value in photometric interpretation, see https://groups.google.com/d/msg/comp.protocols.dicom/KTZiZ-XnYLA/kpD36C2af6UJ

Transforming YBR_FULL_422 into YBR_FULL seems the only possible correction. The result of an original RGB image would remain with false colors. Unfortunately, in this case, the only real solution would be to apply a patch on these incorrect images.

**Comment by nroduit** [comment]
However from a data compression point of view, producing lossy jpeg images with an RGB model remains an absurdity even if it is not forbidden in DICOM.

**Comment by wigun** [comment]
looks like this is fixed by https://github.com/dcm4che/dcm4che/issues/999 ?!

**Comment by nroduit** [comment]
No, it is not related with OpenCV decompressors because only one data model (RGB) is returned for all the image types. This is related to other decompressors that handle all the data models defined in JAI.

**Comment by bcode2** [comment]
Any update?. The problem still persists :(
